lib/OpenQA/Worker/Settings.pm

```diff
@@ -83,6 +84,15 @@ sub new ($class, $instance_number = undef, $cli_options = {}) {
     return $self;
 }
 
+sub auto_detect_worker_address ($self, $fallback = undef) {
```

Comment:
Considering this is more a computed property, maybe a different name would be more descriptive? Like `auto_detected_worker_address` maybe?

Reply:
If `WORKER_HOSTNAME` was not explicitly set, this function will always auto-detect it (again). So the function name doesn't lie. The idea is to avoid getting stuck on a "bad" value. Note that this function does *not* return the address, only whether the auto-detection worked (or `WORKER_HOSTNAME` was set explicitly).

Reply:
Oops, i missed the return value.